packages/suite-desktop/src/support/screens/TorLoadingScreen.tsx

```diff
@@ -0,0 +1,261 @@
+
+const Wrapper = styled.div`
+    height: 100%;
+    display: flex;
+    flex-direction: column;
+    align-items: center;
+    justify-content: center;
+    gap: 16px;
+`;
+
+const Text = styled(P)`
+    height: 0;
+    font-size: ${variables.FONT_SIZE.H2};
+    font-weight: ${variables.FONT_WEIGHT.MEDIUM};
+    color: ${props => props.theme.TYPE_DARK_GREY};
+`;
+
+const ModalWindow = styled.div`
+    display: flex;
+    flex-direction: column;
+    position: relative;
+    border-radius: 16px;
+    text-align: center;
+    transition: all 0.3s;
+    max-width: 95%;
+    min-width: 305px;
+    width: 720px;
+    padding: 42px;
+
+    ${({ theme }) =>
+        css`
+            background: ${theme.BG_WHITE};
+            box-shadow: 0 10px 80px 0 ${theme.BOX_SHADOW_MODAL};
+        `}
+`;
+
+const Percentage = styled.div`
+    display: flex;
+    font-variant-numeric: tabular-nums;
+    height: 24px;
+`;
+
+const StyledProgress = styled(Progress)`
+    display: flex;
+    margin-right: 20px;
+    margin-left: 20px;
+    border-radius: 5px;
+    background: ${({ theme }) => theme.STROKE_GREY_ALT};
+    flex: 1;
+
+    ${Progress.Value} {
+        height: 3px;
+        position: relative;
+        border-radius: 5px;
+    }
+`;
+
+const ProgressWrapper = styled.div`
+    display: flex;
+    align-items: center;
+    background: #e8e8e8;
+    border-radius: 8px;
+    width: 100%;
+    min-height: 45px;
+`;
+
+const ProgressMessage = styled.div`
+    margin-left: 10px;
+    margin-right: 10px;
+    color: ${props => props.theme.TYPE_DARK_GREY};
+`;
+
+const InfoWrapper = styled.div`
+    display: flex;
+    justify-content: space-between;
+`;
+
+const MessageWrapper = styled.div`
+    display: flex;
+    flex-direction: column;
+    align-items: center;
+    justify-content: center;
+    padding-bottom: 50px;
+`;
+
+const Separator = styled.hr`
+    height: 1px;
+    width: 100%;
+    background: none;
+    border: 0;
+    border-top: 1px solid ${props => props.theme.STROKE_GREY};
+    margin: 20px;
+`;
+
+const TryAgainWrapper = styled.div`
+    display: flex;
+    flex-direction: column;
+    align-items: center;
+`;
+
+const DisableButton = styled(Button)`
+    padding-right: 30px;
+    padding-left: 30px;
+    border: 1px solid #f4f4f4;
+    margin-left: 10px;
+`;
+
+interface TorLoadingScreenProps {
+    callback?: (value?: unknown) => void | null;
+    torSettings: { running: boolean; address: string };
+}
+
+export const TorLoadingScreen = ({ callback, torSettings }: TorLoadingScreenProps) => {
+    const [torStatus, setTorStatus] = useState<TorStatus>(TorStatus.Enabling);
+    const [progress, setProgress] = useState<number>(0);
+
+    const setError = () => {
+        setTorStatus(TorStatus.Error);
+    };
+
+    useEffect(() => {
+        if (callback) {
+            setError();
+        }
+    }, [callback]);
+
+    // We unsubscribe and subscribe again, to avoid memory leaks.
+    desktopApi.removeAllListeners('tor/bootstrap');
```

Comment:
I think listeners should be handled inside `useEffect` like
```
useEffect(() => {
    x.addListener(y);
    return () => x.removeListener(y);
}, []);
```
Without that, a) redundant executions may occur, and b) the last executed action is `x.addListener` without corresponding `x.removeListener`.

Reply:
:top: :tophat: e82b02ca5f4f4187331260658bc114581fbad074